pkg/states/state_manager.go

```diff
@@ -0,0 +1,83 @@
+package states
+
+import "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+func Active(userSignup *v1alpha1.UserSignup) bool {
+	return Approved(userSignup) &&
+		!VerificationRequired(userSignup) &&
+		!Deactivated(userSignup)
+}
+
+func Approved(userSignup *v1alpha1.UserSignup) bool {
+	return contains(userSignup.Spec.States, v1alpha1.UserSignupStateApproved)
+}
```

Comment:
btw, shouldn't be the function of the name rather `IsApproved` ?

Reply:
I'm not so sure that `Active` is the same as `Approved`.  We have a well established concept of an "active" user, for a couple of examples see [here](https://github.com/codeready-toolchain/registration-service/blob/master/pkg/kubeclient/signup.go#L90) or [here](https://github.com/codeready-toolchain/api/blob/master/pkg/apis/toolchain/v1alpha1/usersignup_types.go#L80), and this function simply enforces this convention.

For the function name, it is inline with the Go language guidelines - see [here](https://golang.org/doc/effective_go#Getters) for a description of getter/accessor naming recommendations.

Reply:
> I'm not so sure that Active is the same as Approved. We have a well established concept of an "active" user, for a couple of examples see here or here, and this function simply enforces this convention.

ok, I wasn't aware of these usages. TBH, I was always trying to avoid using the "active" word in the code, since it doesn't really represent the "active user", but rather an "approved and provisioned user". I would understand (with my poor English :wink:) "active user" as a user who is logged in to Dev Sandbox.

My point was that we use the `approved` word in:
* usersignup spec
* as a reason in `Completed` condition type
* in UserSignup controller to check if user is provisioned
* in sandbox-cli commands
* and mainly also in our metrics

So I thought that it would be fine to use the same word also for the name of this function. WDYT @alexeykazakov ?

Btw - I just shared my point of view and my feeling, but I don't insist on changing it. If you agree that using `Active` is fine, then I'm fine too and I'll get used to it :wink: 

> For the function name, it is inline with the Go language guidelines - see here for a description of getter/accessor naming recommendations.

thanks for the link :+1:

Reply:
@MatousJobanek you brought a legit concern and we have discussed it with Shane. We currently have a two very similar but sometimes confusing notions/states: approved and active.
What we want to have is some definition of a desired state which will tell the controller that everything is good (from the signup point of view) and we want to provision the account.
Do we call it "approved", "active", or something else is an open question.
But since we do not plan to use this particular state(s) right away and the only state which will be used right now is "deprovisioning" we agreed on leaving it as is. As a separate task we will review our approved/active state notions and since we will need to refactor our controllers (and cli) to migrate to the new Spec.State for provisioning new users anyway we do not have to solve this problem right now.
Bottom line. Let's keep it as is and have a detailed discussion when we ready to start working on the Spec.Approved/Deactivated controller refactoring.